src/hooks/Trades.ts

```diff
@@ -1,26 +1,38 @@
 import { useMemo } from 'react'
-import { WETH, Token, TokenAmount, Trade } from '@uniswap/sdk'
+import { WETH, Token, TokenAmount, Trade, ChainId } from '@uniswap/sdk'
 import { useWeb3React } from './index'
-import { usePair } from '../contexts/Pairs'
-import { isWETH } from '../utils'
+import { useReserves } from '../data/Reserves'
 
 /**
  * Returns the best trade for the exact amount of tokens in to the given token out
  */
+const DAI = new Token(ChainId.MAINNET, '0x6B175474E89094C44Da98b954EedeAC495271d0F', 18, 'DAI', 'Dai Stablecoin')
+const USDC = new Token(ChainId.MAINNET, '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48', 6, 'USDC', 'USD//C')
 export function useTradeExactIn(amountIn?: TokenAmount, tokenOut?: Token): Trade | null {
+  const inputToken = amountIn?.token
+  const outputToken = tokenOut
+
   const { chainId } = useWeb3React()
 
   // check for direct pair between tokens
-  const pairBetween = usePair(amountIn?.token, tokenOut)
+  const pairBetween = useReserves(inputToken, outputToken)
 
   // get token<->WETH pairs
-  const aToETH = usePair(amountIn && !isWETH(amountIn.token) ? amountIn.token : null, WETH[chainId])
-  const bToETH = usePair(tokenOut && !isWETH(tokenOut) ? tokenOut : null, WETH[chainId])
+  const aToETH = useReserves(inputToken, WETH[chainId])
+  const bToETH = useReserves(outputToken, WETH[chainId])
+
+  // get token<->DAI pairs
+  const aToDAI = useReserves(inputToken, chainId === ChainId.MAINNET ? DAI : null)
+  const bToDAI = useReserves(outputToken, chainId === ChainId.MAINNET ? DAI : null)
+
+  // get token<->USDC pairs
+  const aToUSDC = useReserves(inputToken, chainId === ChainId.MAINNET ? USDC : null)
+  const bToUSDC = useReserves(outputToken, chainId === ChainId.MAINNET ? USDC : null)
 
   return useMemo(() => {
-    const allPairs = [pairBetween, aToETH, bToETH].filter(p => !!p)
+    const allPairs = [pairBetween, aToETH, bToETH, aToDAI, bToDAI, aToUSDC, bToUSDC].filter(p => !!p)
```

Comment:
optionally, usdc to dai? dai to eth? usdc to eth?

Reply:
A -> USDC -> DAI -> B seems like a bad option, but in theory yes we could include them. perhaps depends on observational data on which pairs have high liquidity.